numpy/core/tests/test_datetime.py

```diff
@@ -171,9 +178,23 @@ def test_nat_argsort_stability(self, size):
                    [-17, 'NaT', -90]], dtype='M8[us]'),
          np.array([[-220, 51, 'NaT'],
                    [-90, -17, 'NaT']], dtype='M8[us]')),
+        # test suite for `timedelta64` type arrays
+        (np.array(['NaT', 1, 2, 3], dtype='m8[ns]'),
+         np.array([1, 2, 3, 'NaT'], dtype='m8[ns]')),
+        # multiple NaTs
+        (np.array(['NaT', 9, 'NaT', -707], dtype='m8[s]'),
+         np.array([-707, 9, 'NaT', 'NaT'], dtype='m8[s]')),
+        # this sort explores another code path for NaT
+        (np.array([1, -2, 3, 'NaT'], dtype='m8[ns]'),
+         np.array([-2, 1, 3, 'NaT'], dtype='m8[ns]')),
+        # 2-D array
+        (np.array([[51, -220, 'NaT'],
+                   [-17, 'NaT', -90]], dtype='m8[us]'),
+         np.array([[-220, 51, 'NaT'],
+                   [-90, -17, 'NaT']], dtype='m8[us]')),
```

Comment:
Since this only uses ints and NaT strings. You could use two paramterizations: One for the dtype (maybe without the unit) and one for the list. And then move the `np.array` call into the test itself. That way you have half the code with the same result.

Reply:
Done. Thanks! 👍